gulpfile.js

```diff
@@ -1405,12 +1412,19 @@ function createTypeScriptDefinitions() {
     .replace(/^(\s*)(export )?const enum (\S+) {(\s*)$/gm, "$1$2enum $3 {$4")
     // Replace JSDoc generation version of defined with an improved version using TS type predicates
     .replace(
-      /defined\(value: any\): boolean/gm,
-      "defined<Type>(value: Type): value is NonNullable<Type>",
+      /\n?export function defined\(value: any\): boolean;/gm,
+      `\n${readFileSync("./packages/engine/Source/Core/defined.d.ts")
+        .toString()
+        .replace(/\n*\/\*.*?\*\/\n*/gms, "")
+        .replace("export default", "export")}`,
```

Comment:
As per my [comment](https://github.com/CesiumGS/cesium/pull/11901#discussion_r1543545338) when I did the same conversion into a `.d.ts` file I would _love_ if we could skip this `replace` and pass the types file into `tsd-jsdoc` but we cannot. I think `defined` and `Check` are some of the only types we really need to override to take advantage of stuff like type predicates but if we add more we should definitely revisit this setup as it will quickly become too unwieldy.